internal/vulnstore/dastore/get.go

```diff
@@ -54,63 +54,161 @@ type Report struct {
+	 wg.Done()
+ 
+  }
+
+
 func get(ctx context.Context, records []*claircore.IndexRecord, opts vulnstore.GetOpts) (map[string][]*claircore.Vulnerability, error) {
 	log := zerolog.Ctx(ctx).With().
-		Str("component", "internal/vulnstore/postgres/get").
+		Str("component", "internal/vulnstore/dastore/get").
 		Logger()
 	ctx = log.WithContext(ctx)
 
-	s1 := "https://f8a-analytics-2445582058137.production.gw.apicast.io/api/v1/component-analyses/pypi/"
-	s2 := "?user_key=9e7da76708fe374d8c10fa752e72989f"
+
+	fmt.Println("Inside DA Store")
+     fmt.Println(len(records))
+
+
+	// s1 := "https://f8a-analytics-2445582058137.production.gw.apicast.io/api/v1/component-analyses/pypi/"
+	// s2 := "?user_key=9e7da76708fe374d8c10fa752e72989f"
 
 	results := make(map[string][]*claircore.Vulnerability)
 	//position := 0
 
-	for _, j := range records {
-		log.Print(j)
-		//dynamically creating the package
-		url := s1 + j.Package.Name + "/" + j.Package.Version + s2
+	 
+	 iterations:=(len(records)/10)
+
+	if (len(records)%10 == 0){
+		iterations=iterations+0
+	}else{
+		iterations=iterations+1
+	}
+//	+((len(records)%10)==0 ? 0 : 1)
+	
+	total_records:=len(records)
 
-		resultjson, _ := http.Get(url)
+	for i:=0;i<iterations;i++{
 
-		body, _ := ioutil.ReadAll(resultjson.Body)
-		fmt.Println(string(body))
-		reports := Report{}
-		err := json.Unmarshal(body, &reports)
+		var req []Request
+	  
+		  start_record:=i*10
+      //    start_record:=0
+		  var end_record int
 
-		if err != nil {
-			fmt.Println(err)
-		}
+		  if total_records>=10{
+			  end_record=start_record+9
+		  }else{
+			  end_record=start_record + total_records -1
+		  }
+
+		  if total_records>=10{
+			  total_records=total_records-10
+	  
+		  }else{
+			  total_records=0
+		  }
+
+		var ids []string
+			 for record:=start_record;record<=end_record;record++{
+
+			 req=append(req,Request{Ecosystem: "pypi",Package: records[record].Package.Name, Version: records[record].Package.Version})   
+			 ids=append(ids,records[record].Package.ID)   
+			  
+		  }
 
-		if len(reports.Result.Recommendation.ComponentAnalysis.Cve) > 0 {
 
-			v := &claircore.Vulnerability{
+			
+			
 
-				ID:                 j.Package.ID,
-				Updater:            "",
-				Name:               reports.Result.Recommendation.ComponentAnalysis.Cve[0].Idd,
-				Description:        reports.Result.Recommendation.Message,
-				Links:              "",
-				Severity:           fmt.Sprint(reports.Result.Recommendation.ComponentAnalysis.Cve[0].Cvss),
-				// NormalizedSeverity: "",
-				FixedInVersion:     reports.Result.Data[0].Cvee.Fixed_in[0],
-				Package: &claircore.Package{ID: "0",
-					Name:    "xyz",
-					Version: "v0.0"},
-				Dist: &claircore.Distribution{},
-				Repo: &claircore.Repository{},
-			}
 
-			//key := strconv.Itoa(position)
 
-			results[j.Package.ID] = append(results[j.Package.ID], v)
+		  var wg sync.WaitGroup
 
-			//position = position + 1
+		  wg.Add(1)
+
+		 go call(req,ids,results,&wg)
+		 
+		 wg.Wait()
```

Comment:
You calling the go routine and synchronising it immediately, this doesn't use the power of concurrency at all. It should be something like below,

```
response := make(chan Response)
for ... {
  go postRequest(req, ch)
}
msg := <-response
.. parse response..
```